array_test.go

```diff
@@ -108,6 +108,61 @@ func TestArray_Constructors(t *testing.T) {
 	})
 }
 
+func TestArray_Decode(t *testing.T) {
+	t.Run("Decode with slice of interface", func(t *testing.T) {
+		target := []interface{}{}
+		testValue := []interface{}{"Foo", 123.0}
+		reporter := newMockReporter(t)
+		arr := NewArray(reporter, testValue)
+		arr.Decode(&target)
+		arr.Equal(target)
+	})
+	t.Run("Decode with slice of struct", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		type S struct {
+			Foo int `json:"foo"`
+		}
+		testValue := []interface{}{
+			map[string]interface{}{
+				"foo": 123,
+			},
+			map[string]interface{}{
+				"foo": 456,
+			},
+		}
+		arr := NewArray(reporter, testValue)
+		target := new([]S)
+		arr.Decode(target)
+		arr.Equal(target)
+	})
+	t.Run("Passing unmarshable value", func(t *testing.T) {
+		reporter := newMockReporter(t)
+		type S struct {
+			Foo int `json:"foo"`
+		}
+		testValue := []interface{}{
+			map[string]interface{}{
+				"foo": 123,
+			},
+			map[string]interface{}{
+				"foo": 456,
+			},
+		}
+		arr := NewArray(reporter, testValue)
+		target := []S{}
+		arr.Decode(target)
```

Comment:
Decide will fail because target is not a pointer. It's not important whether it's using struct or not.

Hence, to make this test easier to understand, let's not use struct here. We can just do something minimal like:

arr.Decode(123)